dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -142,6 +143,97 @@ static ClassDumper create(List<Path> jarFilePaths) {
+        symbolTableBuilder.classReferencesBuilder();
+
+    String sourceClassName = javaClass.getClassName();
+    ConstantPool constantPool = javaClass.getConstantPool();
+    Constant[] constants = constantPool.getConstantPool();
+    List<Constant> methodrefConstants = Arrays.stream(constants)
+        .filter(Predicates.notNull())
+        .collect(Collectors.toList());
+    for (Constant constant : methodrefConstants) {
+      byte constantTag = constant.getTag();
+      switch (constantTag) {
+        case Const.CONSTANT_Methodref:
+          ConstantMethodref constantMethodref = (ConstantMethodref) constant;
+          methodReferences.add(constantToMethodReference(constantMethodref, constantPool,
+              sourceClassName));
+          break;
+        case Const.CONSTANT_Fieldref:
+          ConstantFieldref constantFieldref = (ConstantFieldref) constant;
+          fieldReferences.add(constantToFieldReference(constantFieldref, constantPool,
+              sourceClassName));
+          break;
+        default:
+          break;
+      }
+    }
+    ImmutableSet.Builder<String> definedClassNameBuilder =
+        symbolTableBuilder.definedClassNamesBuilder();
+    definedClassNameBuilder.add(sourceClassName).addAll(listInnerClassNames(javaClass));
+
+    return symbolTableBuilder.build();
+  }
+
+  static MethodSymbolReference constantToMethodReference(ConstantMethodref constantMethodref,
+      ConstantPool constantPool, String sourceClassName) {
+    String classNameInMethodReference = constantMethodref.getClass(constantPool);
+    int nameAndTypeIndex = constantMethodref.getNameAndTypeIndex();
+    Constant constantAtNameAndTypeIndex = constantPool.getConstant(nameAndTypeIndex);
+    if (!(constantAtNameAndTypeIndex instanceof ConstantNameAndType)) {
+      // This constant_pool entry must be a CONSTANT_NameAndType_info
+      // as specified https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.4.2
+      throw new RuntimeException(
+          "Failed to lookup nameAndType constant indexed "
+              + nameAndTypeIndex
+              + ". This class file is not compliant with CONSTANT_Methodref_info specification");
+    }
+    ConstantNameAndType constantNameAndType = (ConstantNameAndType) constantAtNameAndTypeIndex;
+    String methodName = constantNameAndType.getName(constantPool);
+    String descriptor = constantNameAndType.getSignature(constantPool);
+    MethodSymbolReference methodReference = MethodSymbolReference.builder()
+        .setSourceClassName(sourceClassName)
+        .setMethodName(methodName)
+        .setTargetClassName(classNameInMethodReference)
+        .setDescriptor(descriptor)
+        .build();
+    return methodReference;
+  }
+
+  static FieldSymbolReference constantToFieldReference(ConstantFieldref constantFieldref,
+      ConstantPool constantPool, String sourceClassName) {
+    // Either a class type or an interface type
```

Comment:
These lines look identical to the start of `constantToMethodReference()`. Can you extract a method that takes a `ConstantCP` and `ConstantPool` and returns a `ConstantNameAndType`, throwing if it can't be cast? Then these two methods will have only what's different between methods and fields.